e2e/libs/csi_volume_snapshot/csi_volume_snapshot.py

```diff
@@ -128,14 +128,17 @@ def cleanup_csi_volume_snapshot_classes(self):
             for item in classes.get("items", []):
                 name = item["metadata"]["name"]
                 logging(f"Deleting csi volume snapshot class {name}")
-                self.api.delete_cluster_custom_object(
-                    group=self.group,
-                    version=self.version,
-                    plural="volumesnapshotclasses",
-                    name=name
-                )
-        except ApiException as e:
-            assert e.status == 404
+                try:
```

Comment:
There are nested `try/catch` blocks in this function. PR https://github.com/longhorn/longhorn-tests/pull/2598 has introduced a `try/catch`.

Reply:
The nested try/catch blocks may help the following case:
1. The func lists all CRs before cleanup. One entry in the list is already in deletion stage
2. The entry disappears, then the func sends a deletion request for it

Without the nested try/catch, the execution will be directly trapped in the exception block then return. The subsequent resources in the list won't be removed.

Reply:
This PR is pending https://github.com/longhorn/longhorn-tests/pull/2604#discussion_r2181468764. Will the teardown function return an error in your current design?

Reply:
Now I am fine with returning an error during teardowm stage. That commit will be dropped anyway. 
If this nested try/catch is not required either, there is no commit in this PR. We can directly close it.

Reply:
I can still find the `TRY/EXCEPT` block in the teardown function of this PR. Could you help update it?

Reply:
Updated. PTAL